file_writer_control/JobStatus.py

```diff
@@ -0,0 +1,101 @@
+from enum import Enum, auto
+from datetime import datetime
+
+
+class JobState(Enum):
+    """
+    The state of a job.
+    """
+    NO_JOB = auto()
+    WAITING = auto()
+    WRITING = auto()
+    TIMEOUT = auto()
+    ERROR = auto()
+    DONE = auto()
+    UNAVAILABLE = auto()
+
+
+class JobStatus:
+    """
+    Contains general information about the (execution) of a job.
+    """
+    def __init__(self, job_id: str):
+        self._job_id = job_id
+        self._service_id = ""
+        self._last_update = datetime.now()
+        self._state = JobState.WAITING
+        self._message = ""
+
+    def update_status(self, new_status):
+        """
+        Updates the status/state of a this instance of the JobStatus class, using another instance.
+        .. note:: The job identifier of this instance and the other must be identical.
+        :param new_status: The other instance of the JobStatus class.
+        """
+        if new_status.job_id != self.job_id:
+            raise RuntimeError(
+                "Job id of status update is not correct ({} vs {})".format(
```

Comment:
Instead of `.format()` may be f-strings should be used.

Reply:
I am/was not familiar with f-strings but it has now been changed.